hcipy/optics/fiber.py

```diff
@@ -178,13 +214,88 @@ def read_out(self):
 		self.intensity = 0
 		return intensity
 
-# This implementation assumes orthogonality of the different fibers.
-# Forward() should be changed if this is added in the future.
-# Also, the modes are independent of wavelength.
+class SingleModeFiberInjection(OpticalElement):
+	'''Injection into a single mode fiber.
+
+	Parameters
+	----------
+	input_grid : Grid
+		The grid of the incoming wavefront.
+	mode : Field generator
+		The mode of the fiber.
+	position : array_like
+		The 2D position of the fiber. If this is None, the fiber will be located at (0, 0).
+	'''
+	def __init__(self, input_grid, mode, position=None):
+		self.input_grid = input_grid
+
+		if position is None:
+			position = np.zeros(2)
+		else:
+			position = np.array(position)
+
+		self.output_grid = CartesianGrid(RegularCoords([1, 1], [1, 1], position))
+		self.output_grid.weights = 1
+
+		# Compute the fiber mode and normalize.
+		self.mode = mode(self.input_grid.shifted(-position))
+		self.mode.grid = self.input_grid
+		self.mode /= np.sqrt(np.sum(np.abs(self.mode)**2 * self.input_grid.weights))
+
+	def forward(self, wavefront):
+		'''Forward propagate the light through the fiber.
+
+		Parameters
+		----------
+		wavefront : Wavefront
+			The incoming wavefront.
+
+		Returns
+		-------
+		Wavefront
+			The complex amplitude of the single-mode fiber.
+		'''
+		output = np.dot(wavefront.electric_field * self.input_grid.weights, self.mode)
```

Comment:
I think you are missing a complex conjugate here. The coupling into a particular mode is,
```
np.conj(E).dot(E_in)
```
This does not matter too much for Gaussian modes because those are completely real. But you do run into issue if we have arbitrary modes.